course3/homework-unique_ptr/test.cc

```diff
@@ -0,0 +1,127 @@
+#include "unique_ptr.h"
+
+#include <cassert>
+#include <iostream>
+#include <memory>
+
+
+struct Number
+{
+	int n;
+	static int ctr;
+	Number(int n) : n(n) { ctr++; }
+	Number(const Number& n) : n(n.n) { ctr++; }
+	~Number() { ctr--; }
+};
+int Number::ctr = 0;
+
+template<typename UniquePtr, UniquePtr (*MakeUnique)(int&&)>
```

Comment:
How can a pass a variadic function here? I tried `template\<typename UniquePtr, UniquePtr (\*MakeUnique)(int&&...)\>`, which does not compile.